db/database.go

```diff
@@ -684,9 +677,15 @@ func (dc *DatabaseContext) TakeDbOffline(reason string) error {
 
 		return nil
 	} else {
-		msg := "Unable to take Database offline, database must be in Online state"
-		base.Infof(base.KeyCRUD, msg)
-		return base.HTTPErrorf(http.StatusServiceUnavailable, msg)
+		// If the DB is already transitioning to: offline or is offline silently return
+		dbState := atomic.LoadUint32(&dc.State)
+		if dbState == DBOffline || dbState == DBResyncing || dbState == DBStopping {
+			return nil
+		}
+
+		msg := "Unable to take Database offline, database must be in Online state but was %s"
```

Comment:
It's possible that this could report "Unable to take Database offline, database must be in Online state but was Online" if the db comes online between line 666 and 681, is that right?  I don't think it merits a rework of the locking, but maybe modify the message in the case where dbState = DBOnline?

Reply:
Ah yeah, good catch :)